ibm/service/power/data_source_ibm_pi_volume_test.go

```diff
@@ -35,3 +36,29 @@ data "ibm_pi_volume" "testacc_ds_volume" {
 }`, acc.Pi_volume_name, acc.Pi_cloud_instance_id)
 
 }
+
+func TestAccIBMPIVolumeDataSource_replication(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPIVolumeDataSourceReplicationConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_pi_volume.testacc_ds_volume", "id"),
+					resource.TestCheckResourceAttr("data.ibm_pi_volume.testacc_ds_volume", "replication_enabled", "true"),
+					resource.TestCheckResourceAttrSet("data.ibm_pi_volume.testacc_ds_volume", "replication_status"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMPIVolumeDataSourceReplicationConfig() string {
```

Comment:
I do not see any difference with respect to previous test case. How the data replication will be different for each case when run together?

Reply:
running both the test case at same time would result in one of the test case failing, because a volume can either replication or non-replication. should I remove the replication test case? consider both test cases scenario wont satisfied at the same time.

Reply:
1. Easy:- You can have a different env variable name for replication enabled volume
2. Have the volume created in the Terraform configuration and read via data source in the same configuration.